SCClassLibrary/Common/GUI/PlusGUI/Math/PlotView.sc

```diff
@@ -327,12 +353,23 @@ Plot {
 	}
 
 	hasSteplikeDisplay {
-		^#[\levels, \steps].includes(plotter.plotMode)
+		^#[\levels, \steps, \bars].includes(plotter.plotMode)
+	}
+
+	needsPenFill {
+		^#[\bars].includes(plotter.plotMode)
 	}
 
 	getIndex { |x|
-		var offset = if(this.hasSteplikeDisplay) { 0.5 } { 0.0 }; // needs to be fixed.
-		^(this.getRelativePositionX(x) - offset).round.asInteger
+		var ycoord = this.dataCoordinates;
+		var xcoord = this.domainCoordinates(ycoord.size);
+		var binwidth = 0;
+		var offset;
+		if (xcoord.size > 0) {
+			binwidth = (xcoord[1] ?? {plotBounds.right}) - xcoord[0];
```

Comment:
you can leave out the trailing semicolons everywhere in `if` statements (it is only a separator).